### PR TITLE
Move parameter configurations

### LINKED ISSUE #5395: Move `parameter_config.py` out of _cwrappers
The code in `parameter_config.py` is not a cwrapper and should be moved somewhere else.

### PR DESCRIPTION
**Pull request by frode-aarstad**
**Issue**
Resolves #5395 

**Approach**
_Short description of the approach_

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Updated documentation
- [ ] Ensured new behaviour is tested

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: .mypy.ini [hunk 1]
```diff
@@ -132,3 +132,7 @@ ignore_missing_imports = True
 
 [mypy-ert.callbacks]
 ignore_errors = True
+
+[mypy-ert.config.*]
+ignore_missing_imports = True
+ignore_errors = True
```
Comment (valentin-krasontovitsch) [inline .mypy.ini:138]: do we have an issue to fix the typing things mypy complains about at a later time? or is it just considered part of continuous efforts / spring cleaning or similar?

File: src/ert/config/summary_config.py [hunk 2]
```diff
@@ -97,6 +99,5 @@ def read_from_file(self, run_path: str, iens: int) -> xr.Dataset:
             coords={"time": axis, "name": keys},
         )
 
-
-def _should_load_summary_key(data_key, user_set_keys):
-    return any(fnmatch(data_key, key) for key in user_set_keys)
+    def _should_load_summary_key(self, data_key: Any, user_set_keys: set[str]) -> bool:
```
Comment (valentin-krasontovitsch) [inline src/ert/config/summary_config.py:102]: it seems like based on the code i'm seeing in ecl for eclsum's [`__iter__`](https://github.com/equinor/ecl/blob/469546ebd1dbf5a56ad9390f47dfde65552114d3/python/ecl/summary/ecl_sum.py#L785) and [`keys`](https://github.com/equinor/ecl/blob/469546ebd1dbf5a56ad9390f47dfde65552114d3/python/ecl/summary/ecl_sum.py#L1513) methods, we can type the `data_key` as a string? although i'm a bit unsure what a `StringList` is :grimacing: is that the reason you used `Any` here? : D

**Review by valentin-krasontovitsch — APPROVED** [review]
looking good!

File: .mypy.ini (hunk 1 above)
Comment (frode-aarstad) [reply .mypy.ini:138]: Its an ongoing effort

**Pull request merged**